### PR TITLE
Split services and collections for OneDrive & SharePoint

### PR DESCRIPTION
**Pull request by meain**
Only code movement, no code changes.

Moved services to `/internal/m365/services/{onedrive,sharepoint,exchange}`
Moved collections to `/internal/m365/collection/{drive,site}`

---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [x] :broom: Tech Debt/Cleanup

#### Issue(s)

* #<issue>

#### Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/collection/site/collection.go [hunk 1]
```diff
@@ -81,6 +81,10 @@ func NewCollection(
 	return c
 }
 
+func (sc *Collection) SetBetaService(betaService *betaAPI.BetaService) {
```
Comment (meain) [inline src/internal/m365/collection/site/collection.go:84]: Added an additional function to help with setting beta service as the field is private.

File: src/internal/m365/collection/site/library_handler.go [hunk 2]
```diff
@@ -9,22 +9,26 @@ import (
...
 
-var _ onedrive.BackupHandler = &libraryBackupHandler{}
+var _ drive.BackupHandler = &libraryBackupHandler{}
 
 type libraryBackupHandler struct {
 	ac    api.Drives
 	scope selectors.SharePointScope
 }
 
+func NewLibraryBackupHandler(ac api.Drives, scope selectors.SharePointScope) libraryBackupHandler {
```
Comment (meain) [inline src/internal/m365/collection/site/library_handler.go:28]: Adding new functions to `libraryBackupHandler`, `libraryRestoreHandler`, `itemBackupHandler` and `itemRestoreHandler`.

File: src/internal/m365/service/sharepoint/backup.go [hunk 3]
```diff
@@ -267,13 +218,63 @@ func collectPages(
...
+		collection.SetBetaService(betaService)
+		collection.AddJob(tuple.ID)
+
+		spcs = append(spcs, collection)
+	}
+
+	return spcs, el.Failure()
+}
+
+func collectLists(
```
Comment (meain) [inline src/internal/m365/service/sharepoint/backup.go:236]: I'm still debating where `collectPages`, `collectLists` and `collectLibraries` should be. Any thoughts?

Comment (ryanfkeepers) [reply src/internal/m365/service/sharepoint/backup.go:236]: Probably in `/m365/collections/site`.

**Review by ryanfkeepers — APPROVED** [review]
Mostly lgtm.  As commented, the library handler needs to move into the collection/drive package.  But otherwise I think we're good.  Approved to unblock, with those changes.

**Pull request merged**